pkg/registry/server/client.go

```diff
@@ -0,0 +1,118 @@
+package server
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/sirupsen/logrus"
+
+	"sigs.k8s.io/yaml"
+
+	"github.com/openshift/ci-tools/pkg/api"
+)
+
+type ResolverClient interface {
+	Config(*api.Metadata) (*api.ReleaseBuildConfiguration, error)
+	ConfigWithTest(base *api.Metadata, testSource *api.MetadataWithTest) (*api.ReleaseBuildConfiguration, error)
+	Resolve([]byte) (*api.ReleaseBuildConfiguration, error)
+}
+
+func NewResolverClient(address string) ResolverClient {
+	return &resolverClient{Address: address}
+}
+
+type resolverClient struct {
+	Address string
+}
+
+func (r *resolverClient) Config(info *api.Metadata) (*api.ReleaseBuildConfiguration, error) {
+	logrus.Infof("Loading configuration from %s for %s", r.Address, info.AsString())
+	req, err := http.NewRequest("GET", fmt.Sprintf("%s/config", r.Address), nil)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create request for configresolver: %w", err)
+	}
+	query := req.URL.Query()
+	query.Add(OrgQuery, info.Org)
+	query.Add(RepoQuery, info.Repo)
+	query.Add(BranchQuery, info.Branch)
+	if len(info.Variant) > 0 {
+		query.Add(VariantQuery, info.Variant)
+	}
+	req.URL.RawQuery = query.Encode()
+	return configFromResolverRequest(req)
+}
+
+func (r *resolverClient) ConfigWithTest(base *api.Metadata, testSource *api.MetadataWithTest) (*api.ReleaseBuildConfiguration, error) {
+	logrus.Infof("Loading configuration from %s for %s", r.Address, base.AsString())
+	req, err := http.NewRequest("GET", fmt.Sprintf("%s/configWithInjectedTest", r.Address), nil)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create request for configresolver: %w", err)
+	}
+	query := req.URL.Query()
+	query.Add(InjectTestQuery, testSource.Test)
```

Comment:
```go
for query, value := range map[string]string {
    InjectTestQuery: testSource.Test,
// ...
} {
    query.add(query, value)
}
```

?

Reply:
I considered that when I wrote it, but then I thought it has no advantages in this specific case. It just makes code more indirect. You'd still need code lines to make that map.

Reply:
Addressed